search/src/java/cz/incad/Kramerius/oai/OaiServlet.java

```diff
@@ -109,16 +109,31 @@ protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws Se
         try {
             sendResponse(oaiWriter, resp);
         } catch (IOException ex) {
-            LOG.log(Level.SEVERE, pid, ex);
+            LOG.log(Level.WARNING, pid + " doesn't exist in Fedora.");
             if (!resp.isCommitted()) {
                 resp.reset();
-                resp.sendError(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
+                sendFakeContent(resp);
             }
         } finally {
             oaiWriter.close();
         }
 
     }
+    
+    private void sendFakeContent(HttpServletResponse resp) {
+        String emptyXml = "<error code=\"404\">";
+        resp.setContentType("text/xml");
+        resp.setCharacterEncoding("UTF-8");
+        resp.setContentLength(emptyXml.length());
+
+        try {
```

Comment:
Zavírat zdroje by se mělo vždy v bloku finally, nebo nejlépe v tomto případě použít try-with-resources (už můžeme konečně používat konstruky z novější Javy - v IDE si to přepni rovnou na Javu 8)